### DISCUSSION TITLE
LilyGO TTGO ESP32 Ethernet + PCI-E new board V1.1 XY_2.14 refuses start Ethernet

### DISCUSSION THREAD

**Discussion by sergshadrap** [arendst/Tasmota#16678] [Q&A] [answered]
Many times sorry to trouble respectful community for the problem several times discussed, but:
Trying to setup Ethernet connection for the LilyGO TTGO ESP32 Ethernet + PCI-E. It's relatively new board with PCI-E extension.
Similar to [https://templates.blakadder.com/lilygo_T-Internet-POE.html](url)
The board refuses to start Ethernet, even the leds doesn't blinking, despite the fact , by loading my own sketch based on eth.h library everything works good. What I've done:
 1. Flashed the last Tasmota firmware
 2. Over console command set up the Template for LiLy TTGO
 3. Check all parameters for Ethernet module manually - Ethernet 1, EthAddress 0 , EthClockMode 1, EthType 0 (8720),  and etc
 4. switch wifi off , reboot and nothing has happened ..
 The Ethernet module works with that parameters
ETH_CLK_MODE    ETH_CLOCK_GPIO0_OUT  
ETH_POWER_PIN   4
ETH_TYPE        ETH_PHY_LAN8720
ETH_ADDR        0
ETH_MDC_PIN     23
ETH_MDIO_PIN    18
Could you give a hint  - what might be wrong in my case? 
PS. Everything works fine over wifi, checked with Wemo emulation module.
I've tried almost all configuration on the forum discussed , no effect.

**Comment by sfromis** [comment]
Since you provided no link or even specs for the product, hard to compare with the parameters you tried with. 

Maybe they depend on some sort of "special sauce" firmware to make it work, but that's just idle speculation, only based on the separation into a PCI-E module.

**Reply by sergshadrap** [reply]
Sorry ,[https://github.com/Xinyuan-LilyGO/T-Internet-COM](https://github.com/Xinyuan-LilyGO/T-Internet-COM)

**Comment by Jason2866** [comment]
Looking [here](https://github.com/Xinyuan-LilyGO/T-Internet-COM/blob/main/example/PlatformIO/FactoryTest/src/Config/config.h)
your config `ETH_ADDR 0` looks wrong and in example sketch there is a power pin set for the LAN8720
```
pinMode(ETH_POWER_PIN, OUTPUT);
  digitalWrite(ETH_POWER_PIN, 1);
```

you have to set in template too. See the example configs https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xdrv_driver/xdrv_82_esp32_ethernet.ino 

EDIT: [The github site differs in the examples what to use. Seems the dont know what the need...](https://github.com/Xinyuan-LilyGO/T-Internet-COM/issues/12)
![T-Internet-COM](https://user-images.githubusercontent.com/24528715/192793311-9fb65627-288b-4618-afa4-a8d847126bdb.jpg)

Maybe you have to set gpio5 to high (3.3V) too.

**Reply by sergshadrap** [reply]
Thank you!  I tried everything you advised, unfortunately nothing has changed. As i understand the absent of leds blinking means no power to 8720 chip and defining of it is the main my task.  I'm little confused, from what point you've decided to use GPIO0 as EMAC_PWR? Look at the schematic [https://github.com/Xinyuan-LilyGO/T-Internet-COM/blob/main/Schematic/LAN8720.pdf](https://github.com/Xinyuan-LilyGO/T-Internet-COM/blob/main/Schematic/LAN8720.pdf) It's exactly shows the GPIO4 as a power pin for 8720 chip? I set up all the "suspicious" GPIO as a relay and tried to switch them on ... no success... Any other idea?

**Comment by Jason2866** [comment]
`GPIO0 as EMAC_PWR` did i? where? Gpio 4 seems the power gpio if the schematic is correct.
Gpio 5 probably can be untouched, as long it does not pull to low (GND). Circuit has a pull up resistor.
Have you setup all that in a template and activated the template "set the tick"?
My suggestion was to use `ETH_ADDR 1`

**Reply by sergshadrap** [reply]
Sorry I misunderstood you, when you write down ETH_ADDR 0. Yep, I tried ETH_ADDR 1 , no success , unfortunately all templates you've mentioned has no ETH_CLK_MODE    ETH_CLOCK_GPIO0_OUT , but I tried different combinations. I don't exactly understand what you mean under "set the tick" , but i setup the templates over console input.. and even restarted the module...

**Reply by sfromis** [reply]
"set the tick" means template activated, you have a checkbox on "Configure Other" where you can copy/paste the template.

**Reply by sergshadrap** [reply]
Does it matter how to config the Templates? I checked all pins are set accordingly to the paper for that board...

**Reply by sfromis** [reply]
As long as you make sure that the correct pin is in the template, **and** you make sure that the template is activated (not just saved), what method you use to set the template (Configure Other, Configure Template or the Template command) makes no difference. Activated template means that Module 0 is selected.

**Reply by Jason2866** [reply]
Just to be 100% sure can you post screenshot(s) of the gpios settings and the activation of this setup?
Tbh, i am out of ideas

**Reply by sergshadrap** [reply]
![console](https://user-images.githubusercontent.com/28298218/193114011-b3ab0835-4cc9-4a43-943a-47a8cdc6df10.jpg)
![console2](https://user-images.githubusercontent.com/28298218/193114056-c539914a-e30c-4147-9943-dddae0cb456d.jpg)

**Reply by sergshadrap** [reply]
And... even I've tried such variation as: setting GPIO4 deliberately in High state, changing  EthAddress to 0,1,3 and etc...
Maybe I'm blind to see something very simple?

**Reply by sfromis** [reply]
You have repeatedly been told to make sure **that your template is activated**. This has not been done, hence it has no effect.

**Reply by sergshadrap** [reply]
I'm newbie , so sorry for that . How could I activate the template?? I set up all the components and pressed "save" button - the device beginning to reboot. Does it mean the Template was being activated or ...?

**Reply by sergshadrap** [reply]
Oh! F....sorry 10000 times , i really missed that small box with activation of the template!

**Reply by Jason2866** [reply]
Does it work now?

**Reply by sergshadrap** [reply]
Thank you, The module has started, unfortunately  started only...   The green led blinking, but the module doesn't see the net.

**Comment by Jason2866** [comment]
Yep, you have to set the clock mode with the command listed here https://tasmota.github.io/docs/Commands/#ethernet
`ETH_ADDR 1` tried?

**Reply by sfromis** [reply]
To be clear, the command is `EthAddress 1`.

**Reply by sergshadrap** [reply]
Sure, and i checked the installed Template in the Configuration menu...
All commands and all statuses have checked in console too...
The main problem - why does the power absent on 8720 pin? I checked pin4 with multimeter - been set to high state

**Comment by sergshadrap** [comment, marked as answer]
Many thanks guys! Everything works now ! The module sees the network.  Changing EthAddress back to 0 makes the things done.
So the module works with these parameters:
The Ethernet module works with that parameters
ETH_CLK_MODE ETH_CLOCK_GPIO0_OUT
ETH_POWER_PIN 4
ETH_TYPE ETH_PHY_LAN8720
ETH_ADDR 0
ETH_MDC_PIN 23
ETH_MDIO_PIN 18
My fault was in missing the Activation box.  Sorry to trouble respectful community.

**Reply by unmesh59** [reply]
I'm trying to get this working too. 

Did you flash the base Tasmota ESP32 image?

Did you have to make any changes to the template at https://templates.blakadder.com/lilygo_T-Internet-POE.html?

The parameters ETH_POWER_PIN, ETH_MDC_PIN and ETH_MDIO_PIN are not described in the Tasmota docs. Did you type them at the console and it worked?

Thanks

**Reply by Jason2866** [reply]
The Tasmota commands to set this options (espressif naming) are here  https://tasmota.github.io/docs/Commands/#ethernet  You have to use accordingly
The others are GPIOs which needs to be assigned in Tasmota configuration.

**Reply by unmesh59** [reply]
Thanks!

**Reply by sergshadrap** [reply]
As for me - more comfortable to set ETH_POWER_PIN, ETH_MDC_PIN and ETH_MDIO_PIN in template accordingly to the pins described by manufacturer's paper. Ethernet parameters easily set up from console. Yep, I've changed the template as described above!

**Reply by sergshadrap** [reply]
Guys, the one thing with activation of the templates is still unclear for me. Suppose I want to change my activated template for Wemo emulation - shall I delete the Template completely , or there are some other pathway to reactivation ?

**Reply by sfromis** [reply]
Activating the template is basically setting Module 0, via the checkbox on "Configure Other", via "Configure Module", or via the Module command. Without Module 0 selected, the template is not in use. No need to delete the old Template text.

However, if you use the `CMND` value in the template, this is not saved after having entered the template in "Configure Other" or the Template command, but instead executed and stripped away.

**Reply by sergshadrap** [reply]
Thank you , if I understand you correctly - the command Module 0 will reactivate all changes in template? - right?
And one more question - Due to have clear result i decided to rebuild whole module de novo.   The problem I tried to eliminate is the module refuses emulate Wemo. Looking at console monitor I noticed a error (as i understand memory violation):
14:32:02.108 -> Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.
14:32:02.108 -> 
14:32:02.108 -> Core  1 register dump:
14:32:02.108 -> PC      : 0x40095a9a  PS      : 0x00060c30  A0      : 0x800960ee  A1      : 0x3ffb26c0  
14:32:02.108 -> A2      : 0xffffffff  A3      : 0x00000017  A4      : 0x00060e20  A5      : 0x3ffbf4cc  
14:32:02.108 -> A6      : 0x003fffff  A7      : 0x00060023  A8      : 0x00000003  A9      : 0xffffffff  
14:32:02.155 -> A10     : 0x00000002  A11     : 0x00000114  A12     : 0x00000017  A13     : 0x0000abab  
14:32:02.155 -> A14     : 0x00060e23  A15     : 0x00060e23  SAR     : 0x00000003  EXCCAUSE: 0x0000001c  
14:32:02.155 -> EXCVADDR: 0x0000001b  LBEG    : 0x4008c2dc  LEND    : 0x4008c2e7  LCOUNT  : 0xffffffff  
14:32:02.155 -> 
14:32:02.155 -> 
14:32:02.155 -> Backtrace:0x40095a97:0x3ffb26c00x400960eb:0x3ffb26e0 0x4009634c:0x3ffb2700 0x40085112:0x3ffb2720 0x40085125:0x3ffb2740 0x40146321:0x3ffb2760 0x40151fbb:0x3ffb2780 0x400e644e:0x3ffb2820 
14:32:02.155 -> 
14:32:02.155 -> 
14:32:02.155 -> 
14:32:02.155 -> 
14:32:02.155 -> ELF file SHA256: 0000000000000000
14:32:02.155 -> 
14:32:02.155 -> Rebooting...
Repeating several times - it might be a reason why the module doesn't emulate a Wemo ...
What is the order of things for such errors ? Can I correct it by myself ?

**Reply by Jason2866** [reply]
What should be controlled via HUE Wemo emulation? Do you have defined a relay or pwm?

**Reply by sergshadrap** [reply]
Relays mostly , but i  would see the restriction....
There are three relays now and one WS2812 led..

**Reply by unmesh59** [reply]
> As for me - more comfortable to set ETH_POWER_PIN, ETH_MDC_PIN and ETH_MDIO_PIN in template accordingly to the pins described by manufacturer's paper. Ethernet parameters easily set up from console. Yep, I've changed the template as described above!

Have you considered contributing your template to https://templates.blakadder.com/new.html ?

**Reply by sergshadrap** [reply]
I don't know... , first of all we still have to figure all mistake out,  EXCVADDR: 0x0000001b tells there are some mistake in code. The second things- I don't see the reason multiplicate logically equal template.  The Chinese manufacturers are so volatile , we can't compete them producing  a templates for every hardware SCU))

**Reply by sergshadrap** [reply]
> What should be controlled via HUE Wemo emulation? Do you have defined a relay or pwm?

Could you advise me ? Shall i start a new discussion for the exception with Wemo emulation? I would try to look for the bug myself, but a huge code here.... Maybe some one could navigate  me out for the right place?
